### PR TITLE
move end to end validation to a dedicated phase

### PR DESCRIPTION
**Pull request by mikedanese**
mostly reshuffling

### CODE REVIEW

**Comment by mikedanese** [comment]
@k8s-bot non-cri e2e test this

**Comment by mikedanese** [comment]
@k8s-bot non-cri e2e test this

**Comment by mikedanese** [comment]
@k8s-bot kops aws e2e test this
@k8s-bot non-cri e2e test this

File: cmd/kubeadm/app/phases/validate/validate.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/api/v1"
+	extensions "k8s.io/client-go/pkg/apis/extensions/v1beta1"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+)
+
+func Validate(kubeconfigPath string) error {
```
Comment (luxas) [inline cmd/kubeadm/app/phases/validate/validate.go]: should have a godoc comment, and I'd prefer if this function just called other private funcs instead.

**Review by luxas — APPROVED** [review]
LGTM, one nit that can be follow-up'd I think.

In the end we might end up with removing all the code you're moving now in favor for more sophisticated tests that indirectly tests the below things as well

**Comment by ivan4th** [comment]
kubeadm-dind-cluster CI started failing on master [like this](https://travis-ci.org/Mirantis/kubeadm-dind-cluster/jobs/218064515#L3234) after merging this PR. `kubeadm init` fails:
```
[apiclient] Created API client, waiting for the control plane to become ready
[apiclient] All control plane components are healthy after 19.541550 seconds
failed to update master node - [no nodes found]
```

**Comment by luxas** [comment]
Thanks @ivan4th, please create an issue in the kubeadm repo

**Comment by jellonek** [comment]
Looks like because of removal in https://github.com/kubernetes/kubernetes/pull/43881/files#diff-8562ce2534384a1bba24e11e71f60cb2L47 - without this wait loop func ends too early, and as it's called in https://github.com/kubernetes/kubernetes/blob/3b1d2343a84d3b8e3fd21554568e06b8594cdf26/cmd/kubeadm/app/cmd/init.go#L215 fail arrive 5 lines below, at the start of `UpdateMasterRoleLabelsAndTaints` during which `findMyself` in  https://github.com/kubernetes/kubernetes/blob/3cc14290ad093d5326128b6244ad94b9ac90a844/cmd/kubeadm/app/phases/apiconfig/setupmaster.go#L39 returns with error.

**Comment by luxas** [comment]
@jellonek Would you be up for making a PR to fix that? Would be awesome!

**Comment by jellonek** [comment]
Yes, I'm working on this right now.

**Pull request merged**